### DISCUSSION TITLE
Problem including Python version and Venv in console_title_template

### DISCUSSION THREAD

**Discussion by Daksol** [JanDeDobbeleer/oh-my-posh#2460] [Troubleshoot] [answered]
Hi
OhMyPosh prompt working fine in Powershell, including displaying the Python version (**.Full**) and Conda virtual environment (**.Venv**) values. Template for that segment is
`template: " \uE235 Py: {{ .Full }} Env: {{ .Venv }}"`

However cannot get the Python version and Venv to show in console_title_template. If I include the Python version or Virtual env tags the displayed console title stays at the default set in relevant Windows Terminal config.

Examples
this works
  `"console_title_template": "{{.Shell}} in {{ .PWD }}",`
while this one, when trying to include the Virtual Envt and Python version, does not
`"console_title_template": "Virt Env {{.Venv}} Py {{ .Full }}",`

thankyou

Environment: Windows 10, Windows Terminal 1.13.x, Powershell 7.2.5, OhMyPosh 7.85.2, Conda

**Comment by JanDeDobbeleer** [comment, marked as answer]
That's not really supported and I need to validate BUT, can you check if you can use [cross segment template properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties)?

**Reply by Daksol** [reply]
I will try as you suggest, and get back to you.

**Reply by Daksol** [reply]
Yeeah!  Dat klopt. The [cross segment template properties](https://ohmyposh.dev/docs/configuration/templates#cross-segment-template-properties) work here

Suppose you have an existing segment in the "prompt" section of the OMP config file - like this
```
{
          "type": "python",
          "style": "powerline",
          "powerline_symbol": "\uE0B0",
          "foreground": "red",
          "background": "yellow",
          "template": " \uE235 Py: {{ .Full }} Env: {{ .Venv }}"
        },
```
Then this code
```  
"console_title_template": "{{.Shell}} {{ if .Segments.Python.Venv }} Conda Py: {{ .Segments.Python.Full }} Env: {{ .Segments.Python.Venv }} {{end}}",
```
will generate (for instance)

- **pwsh** when Conda is not running
- **pwsh Py: 3.9.10 Env: xeus** when Conda is running

---

### Further note for configuring Conda to work with OMP
Supplement notes for the [OMP Python segment](https://ohmyposh.dev/docs/segments/python)

Normal behaviour for Conda is that the prompt is modified by including a prefix consisting of the virtual environment name within parenthesis. However if you have configured OMP to put the virtual envt name into the OMP prompt then that standard Conda behaviour no longer needed.

This changeps1 setting is persistent - only need to do this once per workstation.

#### To view the current setting

- At Conda command line, `conda config --show changeps1`
- That will give "changeps1: True" or "changeps1:False"
- With OMP including a Conda env you need changeps1 to be False

#### To switch off or on that default behaviour - two methods

##### At command line Switch off
 To switch off that default behaviour
- When running conda issue this command line
` conda config --set changeps1 False`   
- and then restart Conda
- when the shell with Conda is restarted the default prompt will be gone

##### At command line Switch on
- same commandline as above but ending True
- and restart the shell with Conda

##### By editing the Conda config file directly:-  .condarc
- at the Shell prompt when running Conda, enter command `conda info` to generate list of conda parameters, settings etc
- note the file path for value "user config file", typically **c:\users\someuserid\\.condarc**
 - quit Conda and edit the Conda config file **".condarc"** - file has no extension
- then add the line `changeps1: false` to switch off

**Reply by JanDeDobbeleer** [reply]
that's amazing really

**Reply by Daksol** [reply]
_The development methodology must have been even more rigorous than you had thought...._

One small point on the documentation paragraph for the console_title_template (c_t_t)

- I was confused originally by their being a "blocks" shown - thinking that this implied that the configuration of the c_t_t could include a "blocks" section (when it does not).
- And I have done a test to see if the position of the c_t_t within the xxx.omp.json config file matters. Answer - it does not seem to, since it still worked when I put c_t_t as the first section below the "$schema"

Thanks for all your work on this, appreciated.

![image](https://user-images.githubusercontent.com/42010857/176928010-2781da48-6782-4ce5-9569-be7d29455dee.png)

**Reply by JanDeDobbeleer** [reply]
@Daksol I guess that's maybe tied to your experience with JSON? As it doesn't matter where the property is at, as long as it's on the correct level. The fact that `blocks` is added is to make it clearer at which level to add `console_title_template`. But you can add that above, below, in between any other property at root level.
